parameter/ArrayParameter.cpp

```diff
@@ -135,13 +106,28 @@ bool CArrayParameter::accessValue(CPathNavigator& pathNavigator, string& strValu
 
         } else {
             // Scalar requested
-            doGetValue(strValue, getOffset() + index * getSize(), parameterAccessContext);
+            CParameter::doGetValue(strValue, getOffset() + index * getSize(), parameterAccessContext);
         }
     }
 
     return true;
 }
 
+/// Actual parameter access
+// String access
+bool CArrayParameter::doSetValue(const string& value, size_t /*offset*/,
+                                 CParameterAccessContext& parameterAccessContext) const
+{
+    return setValues(0, parameterAccessContext.getBaseOffset(), value, parameterAccessContext);
```

Comment:
The base offset has been introduced to allow reusing access methods for both main and configuration blackboard.
Here there's a regression due to interface mismatch between doSetValue and setValue in Array parameter class